### ISSUE TITLE
Needs tests

### ISSUE THREAD

**Issue by julie777** [crdoconnor/strictyaml#168]
Feel free to raise an issue for any of the following:

* Feature request

There are no tests in the repository

**Comment by jnichols0** [comment]
The tests are in the hitch directory, you can read about running them here https://hitchdev.com/approach/contributing/

**Comment by julie777** [comment]
> The tests are in the hitch/story directory, you can read about running them here https://hitchdev.com/approach/contributing/

Thanks. I never thought of looking in a different repo for them.

I am going to need to study hitchstory. I have substantial experience using behave for BDD. I am curious to see the changes.

**Comment by mtelka** [comment]
Is there a simple way how to run tests for this project?  Thank you.

**Comment by crdoconnor** [comment]
Yes, apologies I let the original instructions get out of date. I'll update
them tonight.

**Comment by crdoconnor** [comment]
The updated instructions are visible here: https://hitchdev.com/approach/contributing/

**Comment by mtelka** [comment]
Thank you.  That looks easy.  If I read the URL properly just `./key.sh make` followed by `./key.sh regression` should be run to test `strictyaml`.  If that's the case then I think you could create simple `tox.ini` file to have both commands run by `tox` which is currently the preferred way how to run tests uniformly for most Python projects.

**Comment by crdoconnor** [comment]
As I understand it tox is just a different way of creating virtual environments, etc. to run tests in. This is something the code run by key.sh already does. It also containerizes the running code to ensure for greater consistency, which tox can't and doesn't do.

**Comment by mtelka** [comment]
Not only that.  `tox` could be used also with current environment (as opposed to virtualized or containerized env) using the `tox-current-env` plugin.  This is exactly what is needed for downstream packagers like me.  I'm packaging `strictyaml` (and many other Python projects) for OpenIndiana and we need to run tests in current environment to make sure the built package will work as expected and designed.

**Comment by crdoconnor** [comment]
ah, fair enough.

if I wrote a separate script that runs the tests in the environment the
script is run in would that suffice?

**Comment by mtelka** [comment]
> if I wrote a separate script that runs the tests in the environment the script is run in would that suffice?

Sounds reasonable.
